diff -up ejabberd-15.10/Makefile.in.makefile ejabberd-15.10/Makefile.in
--- ejabberd-15.10/Makefile.in.makefile	2015-10-28 16:53:12.000000000 +0200
+++ ejabberd-15.10/Makefile.in	2015-11-11 17:28:06.126369485 +0200
@@ -18,13 +18,13 @@ BINDIR = $(DESTDIR)@bindir@
 SBINDIR = $(DESTDIR)@sbindir@
 
 # /lib/
-LIBDIR =  $(DESTDIR)@libdir@
+LIBDIR =  $(DESTDIR)@libdir@/ejabberd
 
 # /lib/ejabberd/
-EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd
+EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd/ejabberd
 
 # /share/doc/ejabberd
-PACKAGE_TARNAME = @PACKAGE_TARNAME@
+PACKAGE_TARNAME = @PACKAGE_TARNAME@-@PACKAGE_VERSION@
 datarootdir = @datarootdir@
 DOCDIR = $(DESTDIR)@docdir@
 
@@ -133,7 +133,7 @@ $(call TO_DEST,$(MAIN_DIRS) $(DEPS_DIRS)
 	$(INSTALL) -d $@
 
 $(call TO_DEST,deps/p1_pam/priv/bin/epam): $(LIBDIR)/%: deps/% $(call TO_DEST,deps/p1_pam/priv/bin/)
-	$(INSTALL) -m 750 $(O_USER) $< $@
+	$(INSTALL) -m 755 $(O_USER) $< $@
 
 copy-files: $(call TO_DEST,$(DEPS_FILES) $(MAIN_FILES))
 
@@ -160,12 +160,12 @@ install: all copy-files
 	#
 	# Administration script
 	[ -d $(SBINDIR) ] || $(INSTALL) -d -m 755 $(SBINDIR)
-	$(INSTALL) -m 550 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
+	$(INSTALL) -m 755 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
 	# Elixir binaries
 	[ -d $(BINDIR) ] || $(INSTALL) -d -m 755 $(BINDIR)
-	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
-	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
-	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
+	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
+	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
+	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
 	#
 	# Init script
 	$(SED) -e "s*@ctlscriptpath@*$(SBINDIR)*" \
@@ -179,9 +179,9 @@ install: all copy-files
 	#
 	# Binary C programs
 	$(INSTALL) -d $(PBINDIR)
-	$(INSTALL) -m 750 $(O_USER) tools/captcha.sh $(PBINDIR)
-	$(INSTALL) -m 750 $(O_USER) tools/joincluster $(PBINDIR)
-	$(INSTALL) -m 750 $(O_USER) tools/leavecluster $(PBINDIR)
+	$(INSTALL) -m 755 $(O_USER) tools/captcha.sh $(PBINDIR)
+	$(INSTALL) -m 755 $(O_USER) tools/joincluster $(PBINDIR)
+	$(INSTALL) -m 755 $(O_USER) tools/leavecluster $(PBINDIR)
 	#
 	# Copy lite.sql
 	[ -d deps/sqlite3 ] && $(INSTALL) -d $(SQLDIR) || true
