diff -up ejabberd-16.08/ejabberdctl.template.ejabberdctl ejabberd-16.08/ejabberdctl.template
--- ejabberd-16.08/ejabberdctl.template.ejabberdctl	2016-08-04 07:49:23.000000000 +0000
+++ ejabberd-16.08/ejabberdctl.template	2016-08-23 15:45:45.954712376 +0000
@@ -7,16 +7,19 @@ ERL_MAX_PORTS=32000
 ERL_PROCESSES=250000
 ERL_MAX_ETS_TABLES=1400
 FIREWALL_WINDOW=""
-ERLANG_NODE=ejabberd@localhost
+ERLANG_NODE=ejabberd@`hostname -s 2>/dev/null || echo "localhost"`
 
 # define default environment variables
 SCRIPT_DIR=`cd ${0%/*} && pwd`
 ERL={{erl}}
 IEX={{bindir}}/iex
 EPMD={{epmd}}
-INSTALLUSER={{installuser}}
+INSTALLUSER=ejabberd
 ERL_LIBS={{libdir}}
 
+# don't create world readable files
+umask 007
+
 # check the proper system user is used if defined
 if [ "$INSTALLUSER" != "" ] ; then
     EXEC_CMD="false"
@@ -27,7 +30,7 @@ if [ "$INSTALLUSER" != "" ] ; then
                 mkdir -p "$INSTALLUSER_HOME"
                 chown "$INSTALLUSER" "$INSTALLUSER_HOME"
             fi
-            EXEC_CMD="su $INSTALLUSER -c"
+            EXEC_CMD="/sbin/runuser -s /bin/bash $INSTALLUSER -c"
         fi
     done
     if [ `id -g` -eq `id -g $INSTALLUSER` ] ; then
@@ -84,9 +87,6 @@ fi
 if [ "$ERLANG_NODE_ARG" != "" ] ; then
     ERLANG_NODE=$ERLANG_NODE_ARG
 fi
-if [ "{{release}}" != "true" -a "$EJABBERD_BIN_PATH" = "" ] ; then
-    EJABBERD_BIN_PATH={{libdir}}/ejabberd/priv/bin
-fi
 EJABBERD_LOG_PATH=$LOGS_DIR/ejabberd.log
 DATETIME=`date "+%Y%m%d-%H%M%S"`
 ERL_CRASH_DUMP=$LOGS_DIR/erl_crash_$DATETIME.dump
