diff -up ejabberd-16.02/configure.ac.configure ejabberd-16.02/configure.ac
--- ejabberd-16.02/configure.ac.configure	2016-02-29 14:24:36.000000000 +0000
+++ ejabberd-16.02/configure.ac	2016-03-02 10:57:20.972281470 +0000
@@ -6,7 +6,8 @@ AC_INIT(ejabberd, 16.02, ejabberd@proces
 REQUIRE_ERLANG_MIN="6.1 (Erlang/OTP 17.1)"
 REQUIRE_ERLANG_MAX="9.0.0 (No Max)"
 
-AC_CONFIG_MACRO_DIR([m4])
+m4_include([m4/ax_lib_sqlite3.m4])
+m4_include([m4/erlang-extra.m4])
 
 # Checks for programs.
 AC_PROG_MAKE_SET
