diff --git a/Makefile.in b/Makefile.in
index eb14749..603ce4f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -24,7 +24,7 @@ LIBDIR =  $(DESTDIR)@libdir@
 EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd
 
 # /share/doc/ejabberd
-PACKAGE_TARNAME = @PACKAGE_TARNAME@
+PACKAGE_TARNAME = @PACKAGE_TARNAME@-@PACKAGE_VERSION@
 datarootdir = @datarootdir@
 DOCDIR = $(DESTDIR)@docdir@
 
@@ -153,13 +153,13 @@ $(sort $(call TO_DEST,$(MAIN_DIRS) $(DEPS_DIRS))):
 	$(INSTALL) -d $@
 
 $(call TO_DEST,deps/p1_pam/priv/bin/epam): $(LIBDIR)/%: deps/p1_pam/priv/bin/epam $(call TO_DEST,deps/p1_pam/priv/bin/)
-	$(INSTALL) -m 750 $(O_USER) $< $@
+	$(INSTALL) -m 755 $(O_USER) $< $@
 
 $(call TO_DEST,priv/sql/lite.sql): sql/lite.sql $(call TO_DEST,priv/sql)
 	$(INSTALL) -m 644 $< $@
 
 $(call TO_DEST,priv/bin/captcha.sh): tools/captcha.sh $(call TO_DEST,priv/bin)
-	$(INSTALL) -m 750 $(O_USER) $< $@
+	$(INSTALL) -m 755 $(O_USER) $< $@
 
 copy-files-sub2: $(call TO_DEST,$(DEPS_FILES) $(MAIN_FILES) priv/bin/captcha.sh priv/sql/lite.sql)
 
@@ -194,12 +194,12 @@ install: all copy-files
 	#
 	# Administration script
 	[ -d $(SBINDIR) ] || $(INSTALL) -d -m 755 $(SBINDIR)
-	$(INSTALL) -m 550 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
+	$(INSTALL) -m 755 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
 	# Elixir binaries
 	[ -d $(BINDIR) ] || $(INSTALL) -d -m 755 $(BINDIR)
-	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
-	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
-	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
+	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
+	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
+	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
 	#
 	# Init script
 	$(SED) -e "s*@ctlscriptpath@*$(SBINDIR)*" \
