diff -up ejabberd-16.08/include/mod_muc_room.hrl.collab ejabberd-16.08/include/mod_muc_room.hrl
--- ejabberd-16.08/include/mod_muc_room.hrl.collab	2016-08-04 07:49:23.000000000 +0000
+++ ejabberd-16.08/include/mod_muc_room.hrl	2016-08-23 15:45:45.954712376 +0000
@@ -50,8 +50,8 @@
     persistent                           = false :: boolean(),
     moderated                            = true :: boolean(),
     captcha_protected                    = false :: boolean(),
-    members_by_default                   = true :: boolean(),
-    members_only                         = false :: boolean(),
+    members_by_default                   = false :: boolean(),
+    members_only                         = true :: boolean(),
     allow_user_invites                   = false :: boolean(),
     allow_subscription                   = false :: boolean(),
     password_protected                   = false :: boolean(),
diff -up ejabberd-16.08/src/ejabberd_http.erl.collab ejabberd-16.08/src/ejabberd_http.erl
--- ejabberd-16.08/src/ejabberd_http.erl.collab	2016-08-04 07:49:23.000000000 +0000
+++ ejabberd-16.08/src/ejabberd_http.erl	2016-08-23 15:45:45.954712376 +0000
@@ -764,7 +764,13 @@ parse_auth(<<"Basic ", Auth64/binary>>) 
         Pos ->
             {User, <<$:, Pass/binary>>} = erlang:split_binary(Auth, Pos-1),
             PassUtf8 = unicode:characters_to_binary(binary_to_list(Pass), utf8),
-            {User, PassUtf8}
+            case str:tokens(User, <<"@">>) of
+                [Nick, Domain] ->
+                    NewUser = <<Nick/binary, "%", Domain/binary>>;
+                _ ->
+                    NewUser = User
+            end,
+            {<<NewUser/binary, "@", (?MYNAME)/binary>>, PassUtf8}
     end;
 parse_auth(<<"Bearer ", SToken/binary>>) ->
     Token = str:strip(SToken),
diff -up ejabberd-16.08/src/extauth.erl.collab ejabberd-16.08/src/extauth.erl
--- ejabberd-16.08/src/extauth.erl.collab	2016-08-04 07:49:23.000000000 +0000
+++ ejabberd-16.08/src/extauth.erl	2016-08-23 15:45:45.954712376 +0000
@@ -31,7 +31,8 @@
 
 -export([start/2, stop/1, init/2, check_password/3,
 	 set_password/3, try_register/3, remove_user/2,
-	 remove_user/3, is_user_exists/2, opt_type/1]).
+	 remove_user/3, is_user_exists/2, opt_type/1,
+	 get_affiliation/4]).
 
 -include("ejabberd.hrl").
 -include("logger.hrl").
@@ -94,6 +95,9 @@ remove_user(User, Server, Password) ->
     call_port(Server,
 	      [<<"removeuser3">>, User, Server, Password]).
 
+get_affiliation(User, Server, Room, RoomServer) ->
+    call_port(?MYNAME, ["getaff", User, Server, Room, RoomServer]).
+
 call_port(Server, Msg) ->
     LServer = jid:nameprep(Server),
     ProcessName = get_process_name(LServer,
@@ -157,7 +161,8 @@ flush_buffer_and_forward_messages(Pid) -
 encode(L) -> str:join(L, <<":">>).
 
 decode([0, 0]) -> false;
-decode([0, 1]) -> true.
+decode([0, 1]) -> true;
+decode(String) -> list_to_atom(String).
 
 opt_type(extauth_instances) ->
     fun (V) when is_integer(V), V > 0 -> V end;
diff -up ejabberd-16.08/src/mod_muc_room.erl.collab ejabberd-16.08/src/mod_muc_room.erl
--- ejabberd-16.08/src/mod_muc_room.erl.collab	2016-08-04 07:49:23.000000000 +0000
+++ ejabberd-16.08/src/mod_muc_room.erl	2016-08-23 15:45:45.954712376 +0000
@@ -1384,43 +1384,15 @@ set_affiliation(JID, Affiliation, StateD
     StateData#state{affiliations = Affiliations}.
 
 get_affiliation(JID, StateData) ->
-    {_AccessRoute, _AccessCreate, AccessAdmin,
-     _AccessPersistent} =
-	StateData#state.access,
-    Res = case acl:match_rule(StateData#state.server_host,
-			      AccessAdmin, JID)
-	      of
-	    allow -> owner;
-	    _ ->
-		LJID = jid:tolower(JID),
-		case (?DICT):find(LJID, StateData#state.affiliations) of
-		  {ok, Affiliation} -> Affiliation;
-		  _ ->
-		      LJID1 = jid:remove_resource(LJID),
-		      case (?DICT):find(LJID1, StateData#state.affiliations)
-			  of
-			{ok, Affiliation} -> Affiliation;
-			_ ->
-			    LJID2 = setelement(1, LJID, <<"">>),
-			    case (?DICT):find(LJID2,
-					      StateData#state.affiliations)
-				of
-			      {ok, Affiliation} -> Affiliation;
-			      _ ->
-				  LJID3 = jid:remove_resource(LJID2),
-				  case (?DICT):find(LJID3,
-						    StateData#state.affiliations)
-				      of
-				    {ok, Affiliation} -> Affiliation;
-				    _ -> none
-				  end
-			    end
-		      end
-		end
-	  end,
-    case Res of
-      {A, _Reason} -> A;
-      _ -> Res
+    User = JID#jid.user,
+    Server = JID#jid.server,
+    Room = StateData#state.room,
+    RoomServer = StateData#state.host,
+    case extauth:get_affiliation(User, Server, Room, RoomServer) of
+        false ->
+            none;
+        Affiliation ->
+            Affiliation
     end.
 
 get_service_affiliation(JID, StateData) ->
@@ -2308,7 +2280,7 @@ send_new_presence1(NJID, Reason, IsIniti
                         children = []}
                 }
         end,
-    Affiliation = get_affiliation(LJID, StateData),
+    Affiliation = member,
     SAffiliation = affiliation_to_list(Affiliation),
     UserList =
         case not (presence_broadcast_allowed(NJID, StateData) orelse
@@ -2409,7 +2381,7 @@ send_existing_presences1(ToJID, StateDat
 		  {FromJID, _} -> ok;
 		  {_, false} -> ok;
 		  _ ->
-		      FromAffiliation = get_affiliation(LJID, StateData),
+		      FromAffiliation = member,
 		      ItemAttrs = case Role == moderator orelse
 				      (StateData#state.config)#config.anonymous
 				      == false
