diff -up ejabberd-16.01/Makefile.in.makefile ejabberd-16.01/Makefile.in
--- ejabberd-16.01/Makefile.in.makefile	2016-01-12 13:07:58.000000000 +0200
+++ ejabberd-16.01/Makefile.in	2016-01-15 15:50:25.242304452 +0200
@@ -18,13 +18,13 @@ BINDIR = $(DESTDIR)@bindir@
 SBINDIR = $(DESTDIR)@sbindir@
 
 # /lib/
-LIBDIR =  $(DESTDIR)@libdir@
+LIBDIR =  $(DESTDIR)@libdir@/ejabberd
 
 # /lib/ejabberd/
-EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd
+EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd/ejabberd
 
 # /share/doc/ejabberd
-PACKAGE_TARNAME = @PACKAGE_TARNAME@
+PACKAGE_TARNAME = @PACKAGE_TARNAME@-@PACKAGE_VERSION@
 datarootdir = @datarootdir@
 DOCDIR = $(DESTDIR)@docdir@
 
@@ -153,13 +153,13 @@ $(sort $(call TO_DEST,$(MAIN_DIRS) $(DEP
 	$(INSTALL) -d $@
 
 $(call TO_DEST,deps/p1_pam/priv/bin/epam): $(LIBDIR)/%: deps/p1_pam/priv/bin/epam $(call TO_DEST,deps/p1_pam/priv/bin/)
-	$(INSTALL) -m 750 $(O_USER) $< $@
+	$(INSTALL) -m 755 $(O_USER) $< $@
 
 $(call TO_DEST,priv/sql/lite.sql): sql/lite.sql $(call TO_DEST,priv/sql)
 	$(INSTALL) -m 644 $< $@
 
 $(call TO_DEST,priv/bin/captcha.sh): tools/captcha.sh $(call TO_DEST,priv/bin)
-	$(INSTALL) -m 750 $(O_USER) $< $@
+	$(INSTALL) -m 755 $(O_USER) $< $@
 
 copy-files-sub2: $(call TO_DEST,$(DEPS_FILES) $(MAIN_FILES) priv/bin/captcha.sh priv/sql/lite.sql)
 
@@ -193,12 +193,12 @@ install: all copy-files
 	#
 	# Administration script
 	[ -d $(SBINDIR) ] || $(INSTALL) -d -m 755 $(SBINDIR)
-	$(INSTALL) -m 550 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
+	$(INSTALL) -m 755 $(G_USER) ejabberdctl.example $(SBINDIR)/ejabberdctl
 	# Elixir binaries
 	[ -d $(BINDIR) ] || $(INSTALL) -d -m 755 $(BINDIR)
-	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
-	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
-	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 550 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
+	[ -f deps/elixir/bin/iex ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/iex $(BINDIR)/iex || true
+	[ -f deps/elixir/bin/elixir ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/elixir $(BINDIR)/elixir || true
+	[ -f deps/elixir/bin/mix ] && $(INSTALL) -m 755 $(G_USER) deps/elixir/bin/mix $(BINDIR)/mix || true
 	#
 	# Init script
 	$(SED) -e "s*@ctlscriptpath@*$(SBINDIR)*" \
