--- rpmbuild/SPECS/ejabberd.spec.orig	2017-06-30 18:46:11.000000000 +0000
+++ rpmbuild/SPECS/ejabberd.spec	2018-03-12 22:24:26.719559860 +0000
@@ -16,8 +16,9 @@
 
 
 Name:           ejabberd
+Epoch:          10
 Version:        15.07
-Release:        4%{?dist}
+Release:        4%{?dist}.collab.1
 Summary:        A distributed, fault-tolerant Jabber/XMPP server
 
 Group:          Applications/Internet
@@ -64,13 +65,15 @@ Patch9: ejabberd-15.07-0009-Fix-permissi
 # Fedora-specific
 Patch10:ejabberd-15.07-0010-Enable-systemd-notification-if-available.patch
 
-Patch11: ejabberd-0011-aarch64.patch
+#Patch11: ejabberd-0011-aarch64.patch
 
 Patch12: ejabberd-14.07-no_guide_html.patch
 
 # Backported from https://github.com/processone/ejabberd/issues/1409
 Patch13: ejabberd-15.07-Fix_migration_of_old_pubsub_db.patch
 
+Patch14: ejabberd-collab.patch
+
 BuildRequires:  erlang
 BuildRequires:  erlang-rebar
 BuildRequires:  expat-devel
@@ -144,7 +147,7 @@ Windows NT/2000/XP).
 %patch10 -p1 -b .systemd_notify
 
 tar xvf %{S:14}
-%patch11 -p1
+#%patch11 -p1
 
 %if ! 0%{?_with_hevea}
 #fixme patch12 -p1
@@ -153,6 +156,8 @@ tar xvf %{S:14}
 # Fix old mnesia db migration
 %patch13 -p1
 
+%patch14 -p1 -b .collab
+
 
 %build
 autoreconf -ivf
