diff -up ejabberd-15.07/ejabberdctl.template.collab ejabberd-15.07/ejabberdctl.template
--- ejabberd-15.07/ejabberdctl.template.collab	2018-01-08 14:58:20.170902942 +0000
+++ ejabberd-15.07/ejabberdctl.template	2018-01-08 15:21:48.202781707 +0000
@@ -7,14 +7,16 @@ ERL_MAX_PORTS=32000
 ERL_PROCESSES=250000
 ERL_MAX_ETS_TABLES=1400
 FIREWALL_WINDOW=""
-ERLANG_NODE=ejabberd@localhost
+ERLANG_NODE=ejabberd@`hostname -s`
 
 # define default environment variables
 SCRIPT_DIR=`cd ${0%/*} && pwd`
 ERL={{erl}}
 IEX={{bindir}}/iex
 EPMD={{bindir}}/epmd
-INSTALLUSER={{installuser}}
+INSTALLUSER=ejabberd
+
+umask 027
 
 # check the proper system user is used if defined
 if [ "$INSTALLUSER" != "" ] ; then
diff -up ejabberd-15.07/src/ejabberd_http.erl.collab ejabberd-15.07/src/ejabberd_http.erl
--- ejabberd-15.07/src/ejabberd_http.erl.collab	2018-01-08 14:58:20.276902538 +0000
+++ ejabberd-15.07/src/ejabberd_http.erl	2018-01-08 15:03:00.403925307 +0000
@@ -756,7 +756,13 @@ parse_auth(<<"Basic ", Auth64/binary>>)
             undefined;
         Pos ->
             {User, <<$:, Pass/binary>>} = erlang:split_binary(Auth, Pos-1),
-            {User, Pass}
+            case str:tokens(User, <<"@">>) of
+                [Nick, Domain] ->
+                    NewUser = <<Nick/binary, "%", Domain/binary>>;
+                _ ->
+                    NewUser = User
+            end,
+            {<<NewUser/binary, "@", (?MYNAME)/binary>>, Pass}
     end;
 parse_auth(<<_/binary>>) -> undefined.
 
diff -up ejabberd-15.07/src/extauth.erl.collab ejabberd-15.07/src/extauth.erl
--- ejabberd-15.07/src/extauth.erl.collab	2015-07-27 09:45:04.000000000 +0000
+++ ejabberd-15.07/src/extauth.erl	2018-01-08 14:59:33.807622039 +0000
@@ -31,7 +31,8 @@
 
 -export([start/2, stop/1, init/2, check_password/3,
 	 set_password/3, try_register/3, remove_user/2,
-	 remove_user/3, is_user_exists/2, opt_type/1]).
+	 remove_user/3, is_user_exists/2, opt_type/1,
+	 get_affiliation/4]).
 
 -include("ejabberd.hrl").
 -include("logger.hrl").
@@ -94,6 +95,9 @@ remove_user(User, Server, Password) ->
     call_port(Server,
 	      [<<"removeuser3">>, User, Server, Password]).
 
+get_affiliation(User, Server, Room, RoomServer) ->
+    call_port(Server, ["getaff", User, Server, Room, RoomServer]).
+
 call_port(Server, Msg) ->
     LServer = jlib:nameprep(Server),
     ProcessName = get_process_name(LServer,
@@ -158,7 +162,8 @@ flush_buffer_and_forward_messages(Pid) -
 encode(L) -> str:join(L, <<":">>).
 
 decode([0, 0]) -> false;
-decode([0, 1]) -> true.
+decode([0, 1]) -> true;
+decode(String) -> list_to_atom(String).
 
 opt_type(extauth_instances) ->
     fun (V) when is_integer(V), V > 0 -> V end;
diff -up ejabberd-15.07/src/mod_muc_room.erl.collab ejabberd-15.07/src/mod_muc_room.erl
--- ejabberd-15.07/src/mod_muc_room.erl.collab	2018-01-08 14:58:20.279902526 +0000
+++ ejabberd-15.07/src/mod_muc_room.erl	2018-01-08 15:09:05.867174201 +0000
@@ -1360,43 +1360,15 @@ set_affiliation(JID, Affiliation, StateD
     StateData#state{affiliations = Affiliations}.
 
 get_affiliation(JID, StateData) ->
-    {_AccessRoute, _AccessCreate, AccessAdmin,
-     _AccessPersistent} =
-	StateData#state.access,
-    Res = case acl:match_rule(StateData#state.server_host,
-			      AccessAdmin, JID)
-	      of
-	    allow -> owner;
-	    _ ->
-		LJID = jlib:jid_tolower(JID),
-		case (?DICT):find(LJID, StateData#state.affiliations) of
-		  {ok, Affiliation} -> Affiliation;
-		  _ ->
-		      LJID1 = jlib:jid_remove_resource(LJID),
-		      case (?DICT):find(LJID1, StateData#state.affiliations)
-			  of
-			{ok, Affiliation} -> Affiliation;
-			_ ->
-			    LJID2 = setelement(1, LJID, <<"">>),
-			    case (?DICT):find(LJID2,
-					      StateData#state.affiliations)
-				of
-			      {ok, Affiliation} -> Affiliation;
-			      _ ->
-				  LJID3 = jlib:jid_remove_resource(LJID2),
-				  case (?DICT):find(LJID3,
-						    StateData#state.affiliations)
-				      of
-				    {ok, Affiliation} -> Affiliation;
-				    _ -> none
-				  end
-			    end
-		      end
-		end
-	  end,
-    case Res of
-      {A, _Reason} -> A;
-      _ -> Res
+    User = JID#jid.user,
+    Server = JID#jid.server,
+    Room = StateData#state.room,
+    RoomServer = StateData#state.host,
+    case extauth:get_affiliation(User, Server, Room, RoomServer) of
+        false ->
+            none;
+        Affiliation ->
+            Affiliation
     end.
 
 get_service_affiliation(JID, StateData) ->
@@ -2147,7 +2119,7 @@ send_new_presence1(NJID, Reason, StateDa
 	   last_presence = Presence}} =
 	(?DICT):find(jlib:jid_tolower(LJID),
 		     StateData#state.users),
-    Affiliation = get_affiliation(LJID, StateData),
+    Affiliation = member,
     SAffiliation = affiliation_to_list(Affiliation),
     SRole = role_to_list(Role),
     lists:foreach(fun ({_LJID, Info}) ->
@@ -2242,8 +2214,7 @@ send_existing_presences1(ToJID, StateDat
 			  case RealToJID of
 			    FromJID -> ok;
 			    _ ->
-				FromAffiliation = get_affiliation(LJID,
-								  StateData),
+				FromAffiliation = member,
 				ItemAttrs = case Role == moderator orelse
 						   (StateData#state.config)#config.anonymous
 						     == false
